src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -1002,6 +1002,71 @@ internal static char ParseFormatSpecifier(ReadOnlySpan<char> format, out int dig
             }
         }
 
+        private static string? FormatBigIntegerToBin(BigInteger value, int digits)
+        {
+            // Get the bytes that make up the BigInteger.
+            byte[]? arrayToReturnToPool = null;
+            Span<byte> bits = stackalloc byte[64]; // arbitrary threshold
+            if (!value.TryWriteOrCountBytes(bits, out int bytesWrittenOrNeeded))
+            {
+                bits = arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bytesWrittenOrNeeded);
+                bool success = value.TryWriteBytes(bits, out bytesWrittenOrNeeded);
+                Debug.Assert(success);
```

Comment:
Does this throw in release builds?

Reply:
Actually here just uses `Debug.Assert(success)` to indicate `value.TryWriteBytes(bits, out bytesWrittenOrNeeded)` will be always successful, no matter debug or release. We have ensured `bits` can cover `value`'s size before.